### PR TITLE
use the dialog's loading indicator

### PR DESCRIPTION
**Pull request by alanrenmsft**
use in-dialog loading indicator 

before
old-loading

after
new-loading

### CODE REVIEW

**Review by kburtram — APPROVED** [review]
LGTM.  Having the dialog blank and the spinner with no text could look a little off if the dialog takes a long time to load.  It may be better UX to have some text in the blank area of the dialog such as "Loading dialog...".

File: extensions/mssql/src/objectManagement/ui/loginDialog.ts [hunk 1]
```diff
@@ -95,31 +95,27 @@ export class LoginDialog extends ObjectManagementDialogBase<ObjectManagement.Log
...
-				this.initializeSqlAuthSection(view);
-				sections.push(this.sqlAuthSection);
-			}
+		const sections: azdata.Component[] = [];
+		this.initializeGeneralSection();
+		sections.push(this.generalSection);
 
-			this.initializeServerRolesSection(view);
-			sections.push(this.serverRoleSection);
+		if (this.isNewObject || this.objectInfo.authenticationType === 'Sql') {
```
Comment (Charles-Gagnon) [inline extensions/mssql/src/objectManagement/ui/loginDialog.ts:102]: We have https://github.com/Microsoft/azuredatastudio/blob/b7e0bbb64a565fd771439403fde9812cc7817fe3/src/sql/azdata.proposed.d.ts#L523 which we should be generally using - but why is the auth type "sql" here and not "SqlLogin"?

File: extensions/mssql/src/objectManagement/ui/loginDialog.ts [hunk 2]
```diff
@@ -155,21 +151,21 @@ export class LoginDialog extends ObjectManagementDialogBase<ObjectManagement.Log
 			this.onObjectValueChange();
 			await this.runValidation(false);
 		});
-		const authTypeContainer = this.createLabelInputContainer(view, localizedConstants.AuthTypeText, this.authTypeDropdown);
+		const authTypeContainer = this.createLabelInputContainer(localizedConstants.AuthTypeText, this.authTypeDropdown);
 
-		this.enabledCheckbox = this.createCheckbox(view, localizedConstants.EnabledText, this.objectInfo.isEnabled);
+		this.enabledCheckbox = this.createCheckbox(localizedConstants.EnabledText, this.objectInfo.isEnabled);
 		this.enabledCheckbox.onChanged(() => {
```
Comment (Charles-Gagnon) [inline extensions/mssql/src/objectManagement/ui/loginDialog.ts:157]: Make sure you go through and add any disposables to a disposable store/registration to ensure they're cleaned up properly.

**Review by Charles-Gagnon — APPROVED** [review]
+1 on what Karl said - the loading indicator is currently too easy to miss and without any loading text doesn't really help the user understand that it's currently loading and not stuck.

**Review by caohai — APPROVED** [review]

File: extensions/mssql/src/objectManagement/ui/loginDialog.ts (hunk 1 above)
Comment (alanrenmsft) [reply extensions/mssql/src/objectManagement/ui/loginDialog.ts:102]: @caohai could you please update STS side to use the strings consistent with the enums defined azdata typings file?

Comment (caohai) [reply extensions/mssql/src/objectManagement/ui/loginDialog.ts:102]: @alanrenmsft Do you mean make the following string changes?
Sql -> SqlLogin
Windows -> Integrated
AAD -> AzureMFA/AzureMFAAndUser

The first two make sense but for AAD, we don't need to distinguish between MFA/MFAAndUser...

**Comment by alanrenmsft** [comment]
I've updated the code to use an in-dialog loading indicator and updated the GIF in the PR description.

File: extensions/mssql/src/objectManagement/ui/loginDialog.ts (hunk 1 above)
Comment (alanrenmsft) [reply extensions/mssql/src/objectManagement/ui/loginDialog.ts:102]: I guess you can leave AAD as is.

Comment (Charles-Gagnon) [reply extensions/mssql/src/objectManagement/ui/loginDialog.ts:102]: Sounds like you'd probably want to make a separate enum then if it's not a direct 1-1 mapping.

**Pull request merged**